library/azure_rm_resource.py

```diff
@@ -260,6 +258,19 @@ def exec_module(self, **kwargs):
 
             if orphan is not None:
                 self.url += '/' + orphan
+
+        # if api_version was not specified, get latest one
+        if not self.api_version:
+            # extract provider and resource type
+            provider = self.url.split("/providers/")[1].split("/")[0]
+            resourceType = self.url.split(provider + "/")[1].split("/")[0]
+            url = "/subscriptions/" + self.subscription_id + "/providers/" + provider
+            api_versions = json.loads(self.mgmt_client.query(url, "GET", { 'api-version': '2015-01-01'}, None, None, [200], 0, 0).text)
+            for rt in api_versions['resourceTypes']:
+                if rt['resourceType'].lower() == resourceType.lower():
```

Comment:
error message on not matching

Reply:
ok, I will fail if api_version not found